File: ShoppingV4UnitTest/app/src/main/java/dk/itu/shoppingv4unittest/ShoppingActivity.java
package dk.itu.shoppingv4unittest;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

public class ShoppingActivity extends AppCompatActivity {
    //Shopping V4 using Fragments and landscape orientation

    //Model: Database of items
    private static ItemsDB itemsDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.shopping);
        itemsDB = ItemsDB.get();
        setUpFragments();
    }

    private void setUpFragments() {
        FragmentManager fm = getSupportFragmentManager();
        Fragment fragmentUI = fm.findFragmentById(R.id.container_ui);
        Fragment fragmentList = fm.findFragmentById(R.id.container_list);
        if ((fragmentUI == null) && (fragmentList == null)) {
            fragmentUI = new UIFragment();
            fragmentList = new ListFragment();
            fm.beginTransaction()
                    .add(R.id.container_ui, fragmentUI)
                    .add(R.id.container_list, fragmentList)
                    .commit();
        }
    }
}

File: ShoppingLiveDataKotlin/app/src/main/java/dk/itu/shoppinglivedata/ShoppingActivity.kt
package dk.itu.shoppinglivedata

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment

class ShoppingActivity : AppCompatActivity() {
    //Shopping LiveData using Fragments and landscape orientation

    //Model: Database of items
    private var itemsDB: ItemsDB? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.shopping)
        itemsDB = ItemsDB.get()
        setUpFragments()
    }

    private fun setUpFragments() {
        val fm = supportFragmentManager
        var fragmentUI: Fragment? = fm.findFragmentById(R.id.container_ui)
        var fragmentList : Fragment? = fm.findFragmentById(R.id.container_list)
        if (fragmentUI == null && fragmentList == null) {
            fragmentUI = UIFragment()
            fragmentList = ListFragment()
            fm.beginTransaction()
                .add(R.id.container_ui, fragmentUI)
                .add(R.id.container_list, fragmentList)
                .commit()
        }
    }
}